clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java

```diff
@@ -412,9 +414,11 @@ private <D> CompletableFuture<D> getFutureCompletedExceptionally(String exceptio
   @Override
   public void close() throws IOException {
     closed = true;
-    producerExecutor.shutdownNow();
+    producerExecutor.shutdown();
```

Comment:
Isn't stopping the immediate stopping of online producers preferred?

Reply:
Personally, I don't know what is the right way to shutdown an `ExecutorService` the right way. I found this link: https://www.baeldung.com/java-executor-service-tutorial#:~:text=To%20properly%20shut%20down%20an,()%20and%20shutdownNow()%20APIs.&text=This%20method%20returns%20a%20list,to%20do%20with%20these%20tasks. and followed that

Reply:
By setting `closed = true`, we also then make sure that no new requests are accepted

Reply:
By setting `closed = true`, we also then make sure that no new requests are accepted

Reply:
Reverted the change. Thanks for the discussion